janitor/functions.py

```diff
@@ -1575,6 +1571,50 @@ def collapse_levels(df: pd.DataFrame, sep: str = "_"):
     return df
 
 
+@pf.register_dataframe_method
+def reset_index_inplace(df: pd.DataFrame, *args, **kwargs):
+    """
+    In Pandas, `reset_index()`, when used in place, does not return a
+    `DataFrame`, preventing this option's usage in the function-chaining
+    scheme. `reset_index_inplace()` provides one the ability to save
+    computation time and memory while still being able to use the chaining
+    syntax core to pyjanitor. This function, therefore, is the chaining
+    equivalent of:
+
+    .. code-block:: python
+
+        df = (
+            pd.DataFrame(...)
+            .operation1(...)
+            .operation2(...)
+        )
+        df.reset_index(inplace=True)
+
+    instead, being called simply as:
+
+    .. code-block:: python
+
+        df = (
+            pd.DataFrame(...)
+            .operation1(...)
+            .operation2(...)
+            .reset_index_inplace()
```

Comment:
I'd suggest swapping `.operation2()` and `.reset_index_inplace()`, in the "before" and "after" just to emphasize that `reset_index_inplace()` is the method-chaining equivalent of `reset_index(inplace=True)`.